beginner_source/torchtext_custom_dataset_tutorial.py

```diff
@@ -0,0 +1,383 @@
+
+
+# %%
+# Let us start by importing required modules:
+
+import torchdata.datapipes as dp
+import torchtext.transforms as T
+import spacy
+from torchtext.vocab import build_vocab_from_iterator
+eng = spacy.load("en_core_web_sm") # Load the English model to be used for tokenizing
+de = spacy.load("de_core_news_sm") # Load the German model to be used for tokenizing
+
+# %%
+# Now we will load the dataset
+
+FILE_PATH = 'data/deu.txt'
+dataPipe = dp.iter.IterableWrapper([FILE_PATH])
+dataPipe = dp.iter.FileOpener(dataPipe, mode='rb')
+dataPipe = dataPipe.parse_csv(skip_lines=0, delimiter='\t', as_tuple=True)
+
+# %%
+# In the above code block, we are doing following things:
+#
+# 1. At line 2, we are creating an iterable of filenames
+# 2. At line 3, we pass the iterable to `FileOpener` which then
+#    opens the file in read mode
+# 3. At line 4, we call a function to parse the file, which
+#    again returns an iterable of tuples representing each rows
+#    of the tab-delimited file
+#
+# Data pipes can be thought of something like a dataset object, on which
+# we can perform various operations.
+# Check `this tutorial <https://pytorch.org/data/beta/dp_tutorial.html>`_ for more details on
+# data pipes.
+#
+# We can verify if the iterable has the pair of sentences as shown
+# below:
+
+for sample in dataPipe:
+    print(sample)
+    break
+
+# %%
+# Note that we also have attribution details along with pair of sentences. We will
+# write a small function to remove the attribution details:
+
+def remove_attribution(row):
+    """
+    Function to keep the first two elements in a tuple
+    """
+    return row[:2]
+dataPipe = dataPipe.map(remove_attribution)
+
+# %%
+# The `map` function at line 2 in above code block can be used to apply some function
+# on each elements of data pipe. Now, we can verify that the data pipe only contains
+# pair of sentences.
+
+
+for sample in dataPipe:
+    print(sample)
+    break
+
+# %%
+# Now, let us define few functions to perform tokenization:
+
+def eng_tokenize(text):
+    """
+    Tokenize an English text and returns list of tokens
+    """
+    return [token.text for token in eng.tokenizer(text)]
+
+def de_tokenize(text):
+    """
+    Tokenize a German text and returns list of tokens
+    """
+    return [token.text for token in de.tokenizer(text)]
+
+# %%
+# Above function accepts a text and returns a list of words
+# as shown below:
+
+print(eng_tokenize("Have a good day!!!"))
+print(de_tokenize("Haben Sie einen guten Tag!!!"))
+
+# %%
+# Building the vocabulary
+# -----------------------
+# Let us consider an English sentence as the source and a German sentence as the target.
+#
+# Vocabulary can be considered as the set of unique words we have in the dataset.
+# We will build vocabulary for both our source and target now.
+#
+# Let us define a function to get tokens from elements of tuples in the iterator.
+# The comments within the function specifies the need and working of it:
```

Comment:
This sentence doesn't add too much value. Let's remove.

Reply:
I have made changes as suggested.